Tests/test_codonalign.py

```diff
@@ -45,18 +44,16 @@ def test_seq(self):
         self.assertEqual(str(codonseq1.get_codon(slice(1, 3))), "TTT---")
         self.assertEqual(str(codonseq1.get_codon(slice(None, None, -1))), "CCCGGATTT---TTTAAA")
 
-        self.assertRaises(ValueError, codonalign.CodonSeq, "AAA-TT")
         self.assertRaises(ValueError, codonalign.CodonSeq, "AAA-T")
-        self.assertRaises(ValueError, codonalign.CodonSeq, "YVVRRDQQQ")
```

Comment:
So do we still have a way to avoid feeding protein sequences to CodonSeq?

Reply:
No there is not. To be consistent with Python's duck-typing principle, `CodonSeq` should assume that the user knows what he is doing, and raise a `ValueError` only if what the user is asking is impossible (for example, if a particular codon is not found in the codon table).

Reply:
Okay. I think `CodonSeq` is primarily used by the `codonalign.build` method and less exposed to the user. So this will be less of a problem.